### DISCUSSION TITLE
Curiosity Nano DB - programmer failure on Linux

### DISCUSSION THREAD

**Discussion by anarchodin** [SpenceKonde/DxCore#96] [Q&A]
I just got an AVR128DB48 Curiosity Nano board, installed DxCore ... and promptly errored out.

I _can_ upload to the board, by using the mass storage device and asking the Arduino IDE to export binaries. But trying to use the Arduino IDE to do the upload consistently gives this warning:

    avrdude: usbdev_open(): WARNING: failed to set configuration 1: Device or resource busy
    avrdude: usbdev_send(): wrote -5 out of 912 bytes, err = Input/output error
    avrdude: jtag3_edbg_prepare(): failed to send command to serial port

This appears to be blocking me from setting any of the fuses, which is a little annoying. (Using the USB drive to flash the device is honestly _better_ for my workflow.) Any idea what might be going on?

**Comment by SpenceKonde** [comment]
I have no experience using these parts on linux. I have not a clue. Could there be some "modem manager" that opens any serial port connected to the computer? do you have appropriate permissions to use serial ports?  does the mass storage need to be unmounted to work? I'm just grasping at straws here, I've barely used the curiosity board, and don't have physical access to any linux systems to test on.

**Comment by SpenceKonde** [comment]
This is not really specific to this core, since that error is coming from avrdude, and would happen independently of my core... Moving to discussions section in hopes of someone else seeing it over there, and because it does not represent a defect in DxCore.

**Reply by anarchodin** [reply]
Fair point. I keep forgetting discussions exist; that's much better suited. Sorry for the noise.

**Reply by microPaul** [reply]
I know very little about Linux, but did you add your Linux userid to the dialout group?  That seems to be necessary to use a serial port, including the USB-UART bridge devices.

**Reply by pfoun** [reply]
For what it is worth, there also must be permission granted in order to use the serial port.  This is done in the /etc/udev/rules.d folder.  With elevated privileges, create a file with a .rules extension.  For example for the curiosity nano, the file could be named 05-CuriosityNano.rules and contain this text:
SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="2175", MODE:="0666"
SUBSYSTEMS=="usb-serial", TAG+="uaccess"

Here is a link to the platformio instructions for this which has rules for a number of boards:
https://docs.platformio.org/en/latest/faq.html#platformio-udev-rules

**Reply by anarchodin** [reply]
These suggestions are credible enough, but are not relevant. I have been using USB serial devices for _years_ on this machine, and the serial device in question is in fact working quite well, when the code on the AVR chip is doing serial. It's just evidently not what Curiosity Nano uses for the programmer interface.

**Reply by pfoun** [reply]
@anarchodin This evening I tried my Curiosity Nano AVR128DB48 boad and I also have been unable to upload to the Curiosity Nano AVR128DB48 with the Arduino IDE in Linux.  Mplabx is also unable to upload to this board in Linux.  In Windows 10 both Mplabx and Arduino IDE upload to the board successfully.

**Comment by SpenceKonde** [comment]
Well if it's failing on MPLab on linux, complain to Microchip support, see
if they've got a fix (and tell us if they do :-P ) Sounds like there's some
driver support type issue to me >.>

On Tue, Apr 27, 2021 at 11:23 PM pfoun ***@***.***> wrote:

> @anarchodin <https://github.com/anarchodin> This evening I tried my
> Curiosity Nano AVR128DB48 boad and I also have been unable to upload to the
> Curiosity Nano AVR128DB48 with the Arduino IDE in Linux. Mplabx is also
> unable to upload to this board in Linux. In Windows 10 both Mplabx and
> Arduino IDE upload to the board successfully.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/DxCore/discussions/96#discussioncomment-667404>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEW4P27YQAGOSN4DC4X3TK55URANCNFSM42C6HLLQ>
> .
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***

**Comment by pr2zi** [comment]
This issue:  https://github.com/MCUdude/MegaCoreX/issues/13 was closed by a update to MegaCoreX. I still have the previous described Error when programming my Curiosity Nano AVR128DB48 with Linux. Did you find a way to fix this?
